base_layer/mmr/benches/smt.rs

```diff
@@ -38,5 +38,57 @@ pub fn benchmark_sparse_merkle_trees(c: &mut Criterion) {
     }
 }
 
-criterion_group!(smt, benchmark_sparse_merkle_trees);
+fn insert_into_smt(keys: &[NodeKey], tree: &mut SparseMerkleTree<Blake2b<U32>>) {
+    keys.iter().for_each(|key| {
+        tree.upsert(key.clone(), ValueHash::default()).unwrap();
+    });
+}
+
+fn delete_from_smt(keys: &[NodeKey], tree: &mut SparseMerkleTree<Blake2b<U32>>) {
+    keys.iter().for_each(|key| {
+        tree.delete(key).unwrap();
+    });
+}
+
+fn time_function(header: &str, f: impl FnOnce()) -> std::time::Duration {
+    println!("Starting: {header}");
+    let now = std::time::Instant::now();
+    f();
+    let t = now.elapsed();
+    println!("Finished: {header} - {t:?}");
+    t
+}
+
+pub fn root_hash(_c: &mut Criterion) {
```

Comment:
Can these operations be refactored to use Criterion's `bench_function` design? It might get a little tricky because of how its iterations work, since you want to work on a fresh tree for each of its iterations to ensure the operations are being run as expected. But at least then you can probably get better timing data.